### DISCUSSION TITLE
VRL: camel case?

### DISCUSSION THREAD

**Discussion by leshow** [vectordotdev/vector#20983] [General]
I see `upcase` and `downcase` functions in VRL. I don't see a `camelcase`. Is this possible or do you need to write your own camel case transformation?

**Comment by jszwedko** [comment]
I think this would be a feature request if you want to open one.

**Reply by leshow** [reply]
ok, no problem. For posterity is there a good way to do this in the meantime?

for instance, I want to translate a `snake_case` to `camelCase`

I could `split`, is there a way to join and capitalize the first letter with VRL?

**Reply by leshow** [reply]
```
      . = map_keys(., true) -> |key| { 
        ret = ""
        s = split(key, "_")
        for_each(s) -> |index, val| {
          if index == 0 {
            ret = join!([ret, val])
          } else {
            ret = join!([ret, join!([upcase(slice!(val, 0, 1)), slice!(val, 1)]) ])
          }
        }
        ret
       }
```
something like this maybe? I have not used VRL extensively. it doesn't appear to work though

**Reply by jszwedko** [reply]
That actually does appear to work. See https://playground.vrl.dev/?state=eyJwcm9ncmFtIjoiLiA9IG1hcF9rZXlzKC4sIHRydWUpIC0%2BIHxrZXl8IHsgXG4gICAgcmV0ID0gXCJcIlxuICAgIHMgPSBzcGxpdChrZXksIFwiX1wiKVxuICAgIGZvcl9lYWNoKHMpIC0%2BIHxpbmRleCwgdmFsfCB7XG4gICAgICAgIGlmIGluZGV4ID09IDAge1xuICAgICAgICAgIHJldCA9IGpvaW4hKFtyZXQsIHZhbF0pXG4gICAgICAgIH0gZWxzZSB7XG4gICAgICAgICAgcmV0ID0gam9pbiEoW3JldCwgam9pbiEoW3VwY2FzZShzbGljZSEodmFsLCAwLCAxKSksIHNsaWNlISh2YWwsIDEpXSkgXSlcbiAgICAgICAgfVxuICAgIH1cbiAgICByZXRcbn0iLCJldmVudCI6eyJodHRwX3N0YXR1cyI6IjIwMCJ9LCJpc19qc29ubCI6ZmFsc2UsImVycm9yIjpudWxsfQ%3D%3D

**Reply by leshow** [reply]
Oh weird, maybe I messed it up in my test
